kernel/boot/Handoff.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file Handoff.cpp
+ * @author Keeton Feave (keetonfeavel@cedarville.edu)
+ * @brief 
+ * @version 0.1
+ * @date 2021-06-02
+ * 
+ * @copyright Copyright the Panix Contributors (c) 2021
+ * 
+ */
+#include <boot/Handoff.hpp>
+#include <lib/stdio.hpp>
+#include <lib/string.hpp>
+#include <mem/paging.hpp>
+#include <dev/serial/rs232.hpp>
+
+#include <multiboot/multiboot2.h>
+#include <stivale/stivale2.h>
+
+namespace Boot {
+
+/*
+ *  ___                   _          __  __
+ * | __| _ __ _ _ __  ___| |__ _  _ / _|/ _|___ _ _
+ * | _| '_/ _` | '  \/ -_) '_ \ || |  _|  _/ -_) '_|
+ * |_||_| \__,_|_|_|_\___|_.__/\_,_|_| |_| \___|_|
+ *
+ */
+
+FramebufferInfo::FramebufferInfo()
+    : _addr(NULL)
+    , _width(0)
+    , _height(0)
+    , _depth(0)
+    , _pitch(0)
+    , _redMaskSize(0)
+    , _redMaskShift(0)
+    , _greenMaskSize(0)
+    , _greenMaskShift(0)
+    , _blueMaskSize(0)
+    , _blueMaskShift(0)
+    , _memoryModel(Undefined_FBMM)
+{
+    // Default constructor.
+}
+
+FramebufferInfo::FramebufferInfo(uint16_t width, uint16_t height, uint16_t depth, uint16_t pitch, void* addr)
+    : _addr(addr)
+    , _width(width)
+    , _height(height)
+    , _depth(depth)
+    , _pitch(pitch)
+{
+    // TODO: Can we call the default constructor or will that overwrite
+    // the values passed in the initialization list?
+    _redMaskSize = 0;
+    _redMaskShift = 0;
+    _greenMaskSize = 0;
+    _greenMaskShift = 0;
+    _blueMaskSize = 0;
+    _blueMaskShift = 0;
+    _memoryModel = Undefined_FBMM;
+}
+
+FramebufferInfo::~FramebufferInfo()
+{
+    // Nothing to deconstruct
+}
+
+uint16_t FramebufferInfo::getWidth() { return _width; }
+uint16_t FramebufferInfo::getHeight() { return _height; }
+uint16_t FramebufferInfo::getDepth() { return _depth; }
```

Comment:
Would it be a good idea to move `FramebufferInfo`'s implementation entirely into the header file so that we don't incur any performance cost by using a class?

Reply:
Yeah, I was trying to figure out how to do that. For some reason the compiler wasn't happy when I tried, so I have to figure that out.

Reply:
Resolved in 6757f8e.